### DISCUSSION TITLE
How to avoid multiple line breaks into lists with white-space: pre-line?

### DISCUSSION THREAD

**Discussion by dryabinin94** [remarkjs/remark#1176] [Q&A]
I have an issue with multiple line breaking symbols in lists. For example I have a text with multiple line breaks like this:
```
 const shortText = '1. First item.\n\n2. Second Item';
```

Also I have class  in css to support line breaks into text (because I need to break the lines into text inside each list item):

```
.test {
   white-space: pre-line
}
```

This is the code which I have:

```
export default function App() {
  const shortText = '1. First item.\n\n2. Second Item';

  return <ReactMarkdown className="test">{shortText}</ReactMarkdown>;
}
```

As a result I got:
![image](https://github.com/remarkjs/remark/assets/70207219/d71c1a42-cf92-4584-88ac-54952a16edb9)

I expect: (+ line break between them)
![image](https://github.com/remarkjs/remark/assets/70207219/bc9a7980-b100-42e0-90ed-f6535466ba3d)

The question is - How to support multiple line breaks into lists and don't break down the marker margins?

Code example: https://stackblitz-starters-z3af3x.stackblitz.io

**Comment by wooorm** [comment]
That’s exactly what `white-space: pre-line` does. Don’t use `white-space: pre-line` if you don’t want `white-space: pre-line`

**Reply by dryabinin94** [reply]
I can't! I need it in my text, I don't understand why it breaks numeration margins. I just want keep it and use some line breaks into text under numeration list.

**Reply by dryabinin94** [reply]
For example this is the case without ReactMarkdown:

![image](https://github.com/remarkjs/remark/assets/70207219/96bf4429-1369-404d-b49a-cb1f896e7dfa)
Result:

![image](https://github.com/remarkjs/remark/assets/70207219/429029f4-703e-4f6c-87e1-d25bf32bfcb1)


We can see some space between list item but they are not moved to new lines.

**Reply by dryabinin94** [reply]
But in ReactMarkdown case:
![image](https://github.com/remarkjs/remark/assets/70207219/528686a9-05f2-4073-ad9c-23df32867ff3)

Result:
![image](https://github.com/remarkjs/remark/assets/70207219/a0641a03-3d3f-4249-b10b-a963d4fb0d9f)

**Reply by wooorm** [reply]
Yes you can. Or you need to explain why you think you can’t.

> I don't understand why it breaks numeration margins.

`white-space: pre-line` styles `\n`. So if you see `\n`, it’s because there is a `\n`.

>  I just want keep it 

No you want other things: https://github.com/remarkjs/.github/blob/main/support.md#asking-quality-questions.

**Reply by dryabinin94** [reply]
Just to mention it happens only if I put double `\n` symbols. Maybe I am not completely understand how `pre-line` works, but I didn't not put `\n\n` before first list item. I put it between first list and second and I don't expect that it has an effect on first item as well. Looks weird.

**Reply by dryabinin94** [reply]
Actually I found the reason.

We have such code example:
![image](https://github.com/remarkjs/remark/assets/70207219/fe31af90-38c1-4ec1-bc1f-16e7f7214201)

Console output:
![image](https://github.com/remarkjs/remark/assets/70207219/176d165f-1eb5-4bd2-bb1a-ad2b48277372)

We can see array of elements and I don't why 0 element is `\n`

**Reply by wooorm** [reply]
Yeas, that’s what I said: https://github.com/orgs/remarkjs/discussions/1176#discussioncomment-6183945.

You’re ignoring my questions: Why do you think you want `pre-line`? Because you probably don‘t need it.

**Reply by dryabinin94** [reply]
Yes, sorry. Actually I have a chat where user can communicate with support. Support has some templates of answer. In one of this templates we have long string with lists, links, emojies e.t.c. To divide parts of message support template has '\n' symbols which I need to handle properly to make text readable as support planed.

**Reply by wooorm** [reply]
That helps a little bit. Please provide more info. See https://github.com/remarkjs/.github/blob/main/support.md#asking-quality-questions for more info. Provide input for example

**Reply by dryabinin94** [reply]
For example - https://stackblitz.com/edit/stackblitz-starters-xrnr2c?file=src%2FApp.tsx

**Reply by wooorm** [reply]
Use markdown. Why are you using markdown if you don’t want to use markdown.

```markdown
1. First paragraph in list item

   Second paragraph in list item.
```

**Reply by wooorm** [reply]
Or do you want soft breaks to turn into hard breaks? Then use https://github.com/remarkjs/remark-breaks.

**Reply by dryabinin94** [reply]
It is no clear for me what do you mean. Do you want me to use markdown syntax in my answers or you tell me about my issue?

**Reply by dryabinin94** [reply]
As for  https://github.com/remarkjs/remark-breaks. it didn't help. I have the same issue.

**Reply by dryabinin94** [reply]
All I want is ReactMarkdown should properly handle such cases which I explained. For me it is like a bug.

**Reply by dryabinin94** [reply]
The particular question is: Why do you add `\n` before list item if it is not in input text. Input - ` const short = '1. Ut enim ad minim veniam \n\n2. Test item'`

**Reply by dryabinin94** [reply]
Also I checked the same situation with another Markdown library - https://www.npmjs.com/package/markdown-to-jsx
And it works good.

**Reply by wooorm** [reply]
> It is no clear for me what do you mean. Do you want me to use markdown syntax in my answers or you tell me about my issue?

I mean: use markdown to *actually* put paragraphs and lists into list items instead. We follow markdown. We work the same as here on GitHub. Or as https://spec.commonmark.org/dingus/.

> it didn't help. I have the same issue.

What did you try? Please provide more info.

> Why do you add \n before list item if it is not in input text.

Because CommonMark requires it.
